src/TcUnit.TestAdapter/Execution/TestRunner.cs

```diff
@@ -137,65 +171,50 @@ private bool IsPlcProjectSuitableForTestRun (PlcProject project)
 
             if (!library.Parameters.TryGetValue("XUNITENABLEPUBLISH", out xUnitEnablePublish))
             {
-                return false;
+                throw NonSuitablePLCProjectException("XUNITENABLEPUBLISH parameter not found");
             }
 
-            if(string.IsNullOrEmpty(xUnitEnablePublish) || !xUnitEnablePublish.Equals("TRUE"))
+            if (string.IsNullOrEmpty(xUnitEnablePublish) || !xUnitEnablePublish.Equals("TRUE"))
             {
-                return false;
+                throw NonSuitablePLCProjectException("XUNITENABLEPUBLISH parameter not set to TRUE");
             }
 
             var xUnitFilePath = "";
 
             if (!library.Parameters.TryGetValue("XUNITFILEPATH", out xUnitFilePath))
             {
-                return false;
+                throw NonSuitablePLCProjectException("XUNITFILEPATH parameter not found");
             }
 
-            if (string.IsNullOrEmpty(xUnitFilePath) )
+            if (string.IsNullOrEmpty(xUnitFilePath))
             {
-                return false;
+                throw NonSuitablePLCProjectException("XUNITFILEPATH parameter not set");
             }
-
-            var useDataTypeAsTestNames = "";
-
-            if (!library.Parameters.TryGetValue("USEDATATYPESASTESTNAMES", out useDataTypeAsTestNames))
-            {
-                return false;
-            }
-
-            if (string.IsNullOrEmpty(useDataTypeAsTestNames) || !useDataTypeAsTestNames.Equals("TRUE"))
-            {
-                return false;
-            }
-
-            return true;
         }
 
-        private void PrepareTargetForTestRun (TargetRuntime target, TwinCATXAEProject project, bool cleanUpBeforeTestRun)
+        private void PrepareTargetForTestRun(TargetRuntime target, TwinCATXAEProject project, bool cleanUpBeforeTestRun)
         {
             target.SwitchToConfigMode();
             target.DownloadProject(project, cleanUpBeforeTestRun);
         }
 
 
-        private IEnumerable<TestResult> PopulateTestRunResults (IEnumerable<TestCase> testCases, IEnumerable<TestCaseResult> testResults)
+        private IEnumerable<TestResult> PopulateTestRunResults(IEnumerable<TestCase> testCases, IEnumerable<TestCaseResult> testResults)
         {
             var testRunResults = new List<TestResult>();
 
             foreach (var test in testCases)
             {
                 var testResult = new TestResult(test);
 
-                var result = testResults.Where(r => r.FullyQualifiedName == test.FullyQualifiedName).First();
+                try {
+                    var result = testResults.Where(r => r.FullyQualifiedName == test.FullyQualifiedName).First();
```

Comment:
This could be improved to not use a try..except. The prior code was causing an exception since Where() would return null (or some other value) that doesn't allow for calling `.First()` on it.